app/controllers/tasks_controller.rb

```diff
@@ -1,8 +1,73 @@
 class TasksController < ApplicationController
+
   def index
+    @filter = params[:category]
+    @all_categories = %w[Inventory Register Engineering]
+    @tasks = if @filter.blank?
+               Task.all
+             else
+               Task.where(category: @filter)
+             end
+    incomplete_task = @tasks.where(completed: false).sort_by { |task| [-task.priority, task.added] }
+    completed_tasks = @tasks.where(completed: true).sort_by { |task| [task.complete_time] }.reverse
+    @tasks = incomplete_task + completed_tasks
+    @num_incomplete = @tasks.select { |task| task.completed == false }.count
+    @count = 0
+    @completed = false
+    @curr_categories = @all_categories
+    @users = User.all
+  end
+
+  def destroy
+    task = Task.find(params[:id])
+    Task.delete(task)
+    redirect_to tasks_path(category: params[:category])
+  end
+
+  def checkmark
+    task = Task.find(params[:id])
+    task.completed = !task.completed
+    user = User.find(params[:user])
+    task.user_complete = user.name
+    task.complete_time = DateTime.now
```

Comment:
prefer `Time.current`, which returns a more usable object and understands timezones